### PR TITLE
Deleting term from trie - fix bug in memcpy upon merging nodes [MOD-5880, MOD-5952]

### PR DESCRIPTION
**Pull request by alonre24**
**Describe the changes in the pull request**

Every index holds a trie data structure that contains the terms that exist in the index (stored as TEXT fields). When forkGC is cleaning deleted documents it also removes terms that are not contained in any document. 

A bug was found in the `Trie_Delete` operation, where we don't copy properly the keys' lookup array upon merging two nodes (we copy only half of the array), since we were copying `numChildren` bytes rather than `numChildren * sizeof(rune)` bytes as we should. Hence, when we looked for a term to delete in the trie, we might have failed to find it (while traversing the tree, at some point we couldn't find the matching key at the lookup), and therefore we didn't delete it. 
This caused the index memory to inflate when performing many updates, and when `maxmemory` configuration was set along with `maxmemory-policy` of `allkeys-lru`, we saw the number of keys decreasing due to multiple updates that triggered evictions which failed to properly clean the memory. Note that deleting the index did reclaim the memory from the trie.

The fix is straightforward - copy the right number of cytes when merging two nodes in the trie after deletion.

**Which issues this PR fixes**
1. MOD-5880

**Main objects this PR modified**
1. The terms trie in index spec

**Mark if applicable**

- [ ] This PR introduces API changes
- [ ] This PR introduces serialization changes

### CODE REVIEW

File: src/fork_gc.c [hunk 1]
```diff
@@ -789,7 +789,11 @@ static FGCError FGC_parentHandleTerms(ForkGC *gc) {
     if (sctx->spec->keysDict) {
       dictDelete(sctx->spec->keysDict, termKey);
     }
-    Trie_Delete(sctx->spec->terms, term, len);
+    int rc = Trie_Delete(sctx->spec->terms, term, len);
+    if (rc == 0) {
```
Comment (GuyAv46) [inline src/fork_gc.c:793]: styling suggestion:
```suggestion
    if (!Trie_Delete(sctx->spec->terms, term, len)) {
```

File: tests/pytests/test_issues.py [hunk 2]
```diff
@@ -975,3 +975,30 @@ def test_mod5910(env):
     env.assertEqual(iterators_profile[1][1], 'INTERSECT')
     env.assertEqual(iterators_profile[1][7][1], 'NUMERIC')
     env.assertEqual(iterators_profile[1][8][1], 'UNION')
+
+
+@skip(cluster)
```
Comment (GuyAv46) [inline tests/pytests/test_issues.py:980]: what does that do?
Did you mean `@skip(cluster=True)` or `@skip(cluster=False)`?

File: tests/pytests/test_issues.py [hunk 3]
```diff
@@ -975,3 +975,30 @@ def test_mod5910(env):
...
+def test_mod5880(env):
+    env.cmd("ft.config", "set", "FORK_GC_CLEAN_THRESHOLD", "0")
+    env.cmd("ft.create", "idx", "schema", "t", "TEXT")
+
+    env.cmd("HSET", "doc1", "t", "d")
+    env.cmd("HSET", "doc2", "t", "dd")
+    env.cmd("HSET", "doc3", "t", "ddd")
+    env.cmd("HSET", "doc4", "t", "dde")
+    env.assertEqual(env.cmd("FT.DEBUG", "dump_terms", "idx"), ['d', 'dd', 'ddd', 'dde'])
```
Comment (GuyAv46) [inline tests/pytests/test_issues.py:989]:
```suggestion
    env.expect("FT.DEBUG", "dump_terms", "idx").equal(['d', 'dd', 'ddd', 'dde'])
```

File: tests/pytests/test_issues.py [hunk 4]
```diff
@@ -975,3 +975,30 @@ def test_mod5910(env):
...
+    # merge operation that didn't copy properly the children keys array ("d" and "e" in our case), so that we only
+    # copied half of the children to the new merged node. Then, when deleting "dde", we couldn't find it in trie, and
+    # was left undeleted (inflating the memory as a consequence).
+    env.cmd("DEL", "doc1")
+    env.cmd("FT.DEBUG", "GC_FORCEINVOKE", "idx")
+
+    # Validate that we actually delete "dde" and that in doesn't exist in the trie.
+    env.cmd("DEL", "doc4")
+    env.cmd("FT.DEBUG", "GC_FORCEINVOKE", "idx")
+    env.assertEqual(env.cmd("FT.DEBUG", "dump_terms", "idx"), ['dd', 'ddd'])
```
Comment (GuyAv46) [inline tests/pytests/test_issues.py:1004]:
```suggestion
    env.expect("FT.DEBUG", "dump_terms", "idx").equal(['dd', 'ddd'])
```

**Review by GuyAv46 — COMMENTED** [review]
No better feeling than fixing an awful bug with a single `*2`

File: tests/pytests/test_issues.py (hunk 2 above)
Comment (oshadmi) [reply tests/pytests/test_issues.py]: Currently to skip also when False, we need to add something with 3-state such as https://github.com/RediSearch/RediSearch/pull/4166/files#diff-48a80e3d857e932ed92f9342791cea987617fa88ca398d0df7b7efffb7972b6fR300

**Review by GuyAv46 — APPROVED** [review]
AMAZING

File: tests/pytests/test_issues.py (hunk 2 above)
Comment (GuyAv46) [reply tests/pytests/test_issues.py]: Ahead of you. On master (soon if not already on 2.6, 2.8) we have it :sunglasses:

**Pull request merged**